source/solvers/free_surface.cc

```diff
@@ -267,6 +305,21 @@ FreeSurface<dim>::assemble_system(
                           cell_matrix(i, j) +=
                             tau * strong_jacobian *
                             (grad_phi_phase_i * velocity_values[q]) * JxW;
+
+                          if (DCDD)
+                            {
+                              cell_matrix(i, j) +=
```

Comment:
why put it in two sets of equations? I guess it is for code clarity, and it's a good idea, but then could you comment on the two equations? (state what they are, or give a reference)

Reply:
Merged into a single equation. It's just two equations because it's the jacobian of a product of two velocity functions.